benchmarks/_benchmark.py

```diff
@@ -52,12 +53,24 @@ def arrow_info() -> Dict[str, Any]:
     }
 
 
+class ConbenchCommunicator(conbench.runner.Conbench):
```

Comment:
Note that I just changed the logic here slightly. Previously, I set up the new `ConbenchClient()` during `__init__()`. But that meant that we tried to login to Conbench upon initialization of any benchmark object in this repo. Besides the fact that this was failing unit tests (which aren't supposed to talk to Conbench at all), this wouldn't be great for users who want to run benchmarks locally but don't have Conbench credentials.

Now we login right before publishing.

Reply:
This arrangement may make for a _lot_ of login calls; something to keep an eye on

Reply:
oof you're right; I thought we did this before this PR, but it looks like the legacy communicator caches the login session. I should fix this. This will be too many calls.